agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/components/AgendaOverviewDateWidget.kt

```diff
@@ -0,0 +1,156 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import com.pronoidsoftware.core.presentation.designsystem.Inter
+import com.pronoidsoftware.core.presentation.designsystem.LocalSpacing
+import com.pronoidsoftware.core.presentation.designsystem.TaskyDarkGray
+import com.pronoidsoftware.core.presentation.designsystem.TaskyGray
+import com.pronoidsoftware.core.presentation.designsystem.TaskyOrange
+import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+import com.pronoidsoftware.core.presentation.ui.endOfMonth
+import com.pronoidsoftware.core.presentation.ui.rangeTo
+import com.pronoidsoftware.core.presentation.ui.startOfMonth
+import com.pronoidsoftware.core.presentation.ui.today
+import java.util.Locale
+import kotlinx.coroutines.launch
+import kotlinx.datetime.LocalDate
+
+@Composable
+fun AgendaOverviewDateWidget(
+    selectedDate: LocalDate,
+    onSelectDate: (LocalDate) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val spacing = LocalSpacing.current
+    val scrollIndex = selectedDate.dayOfMonth - 1
+    val offsetPx = with(LocalDensity.current) { spacing.agendaDateWidgetOffset.roundToPx() }
+    val rowState = rememberLazyListState(
+        initialFirstVisibleItemIndex = scrollIndex,
+        initialFirstVisibleItemScrollOffset = scrollIndex - offsetPx,
+    )
+    val coroutineScope = rememberCoroutineScope()
+    LaunchedEffect(selectedDate) {
+        rowState.animateScrollToItem(
+            index = scrollIndex,
+            scrollOffset = scrollIndex - offsetPx,
+        )
+    }
+    val startDate = selectedDate.startOfMonth()
+    val endDate = selectedDate.endOfMonth()
+    val range = startDate.rangeTo(endDate).toList()
+    LazyRow(
+        modifier = modifier
+            .fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(spacing.agendaItemPaddingHorizontal),
+        state = rowState,
+    ) {
+        items(
+            items = range,
+            key = { it.dayOfYear },
+        ) { date ->
+            DateDisplay(
+                date = date,
+                selected = date.dayOfMonth == selectedDate.dayOfMonth,
+            ) { newSelectedDate ->
+                if (newSelectedDate == selectedDate) {
+                    coroutineScope.launch {
+                        rowState.animateScrollToItem(
+                            index = scrollIndex,
+                            scrollOffset = scrollIndex - offsetPx,
+                        )
+                    }
+                } else {
+                    onSelectDate(newSelectedDate)
+                }
+            }
```

Comment:
In Compose, I'd be careful with trailing non-Compose lambdas, since they look so much like a side-effect 😅 

My first thought was that this `DateDisplay` has some kind of dynamic composable content, but only at the second glance it shows that it's probably a lambda for when the date is selected. Instead, I personally always use named parameters for normal lambdas

Reply:
good catch, I usually try to follow the same but missed it here